#! /bin/sh -e 
## <PATCHNAME>.dpatch by <PATCH_AUTHOR@EMAI>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Description: [CAN-2005-0204]: AMD64, allows local users to write to privileged IO ports via OUTS instruction
## DP: Patch author: Suresh Siddha (suresh.b.siddha@intel.com)
## DP: Upstream status: unknown
## DP: URL: https://bugzilla.redhat.com/bugzilla/show_bug.cgi?id=146244
## DP: Patch source: Micah Anderson <micah@riseup.net> (debian-kernel)

. $(dirname $0)/DPATCH

@DPATCH@
--- linux-2.6.9/include/asm-x86_64/desc.h	2005-01-30 20:08:12.799247944 -0800
+++ linux-2.6.9/include/asm-x86_64/desc.h	2005-01-30 20:08:12.799247944 -0800
@@ -128,7 +128,7 @@
 { 
 	set_tssldt_descriptor(&cpu_gdt_table[cpu][GDT_ENTRY_TSS], (unsigned long)addr, 
 			      DESC_TSS,
-			      sizeof(struct tss_struct) - 1);
+			      IO_BITMAP_OFFSET + IO_BITMAP_BYTES + 7);
 } 
 
 static inline void set_ldt_desc(unsigned cpu, void *addr, int size)
